Restructure Wasp Deploy docs (#2957)
wasp-lang/wasp: web/docs/general/cli.md

- `wasp deploy` makes it easy to get your app hosted on the web.

> Currently, Wasp offers support for [Fly.io](https://fly.io) and [Railway](https://railway.com/?utm_medium=integration&utm_source=docs&utm_campaign=wasp). If you prefer a different hosting provider, feel free to let us know on Discord or submit a PR by updating [this TypeScript app](https://github.com/wasp-lang/wasp/tree/main/waspc/packages/deploy).

Reviewer: Just a thought (maybe it's for a different PR) -- maybe it makes sense to configure this at a global level and inject the parameters, instead of littering them throughout our docs? (idk if it's possible tho)

Author: I'd say - let the Railway team maintain this? If we go too fancy with this it might harder for them?

> Currently, Wasp offers support for [Fly.io](https://fly.io) and [Railway](https://railway.com/?utm_medium=integration&utm_source=docs&utm_campaign=wasp). If you prefer a different hosting provider, feel free to let us know on Discord or submit a PR by updating [this TypeScript app](https://github.com/wasp-lang/wasp/tree/main/waspc/packages/deploy).

Reviewer: ```suggestion
Currently, Wasp offers support for [Fly.io](https://fly.io) and [Railway](https://railway.com/?utm_medium=integration&utm_source=docs&utm_campaign=wasp). If you prefer a different hosting provider, you can [let us know on Discord](https://discord.gg/rzdnErX) or [contribute the code yourself](https://github.com/wasp-lang/wasp/tree/main/waspc/packages/deploy).
```
